pkg/services/grafana-apiserver/storage/entity/fieldRequirements.go

```diff
@@ -5,46 +5,50 @@ import (
 	"strings"
 
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/selection"
 )
 
 const folderAnnoKey = "grafana.app/folder"
+const sortByKey = "grafana.app/sortBy"
 
 type FieldRequirements struct {
 	// Equals folder
 	Folder *string
+	// SortBy is a list of fields to sort by
+	SortBy []string
 }
 
-func ReadFieldRequirements(selector fields.Selector) (FieldRequirements, fields.Selector, error) {
+func ReadFieldRequirements(selector labels.Selector) (FieldRequirements, labels.Selector, error) {
 	requirements := FieldRequirements{}
+	newSelector := labels.NewSelector()
 
 	if selector == nil {
-		return requirements, selector, nil
+		return requirements, newSelector, nil
 	}
 
-	for _, r := range selector.Requirements() {
-		switch r.Field {
+	labelSelectors, _ := selector.Requirements()
+
+	for _, r := range labelSelectors {
+		switch r.Key() {
 		case folderAnnoKey:
-			if (r.Operator != selection.Equals) && (r.Operator != selection.DoubleEquals) {
-				return requirements, selector, apierrors.NewBadRequest("only equality is supported in the selectors")
+			if (r.Operator() != selection.Equals) && (r.Operator() != selection.DoubleEquals) {
+				return requirements, newSelector, apierrors.NewBadRequest(folderAnnoKey + " label selector only supports equality")
 			}
-			folder := r.Value
+			folder := r.Values().List()[0]
 			requirements.Folder = &folder
+		case sortByKey:
+			if r.Operator() != selection.In {
+				return requirements, newSelector, apierrors.NewBadRequest(sortByKey + " label selector only supports in")
+			}
+			requirements.SortBy = r.Values().List()
+		default:
+			newSelector = newSelector.Add(r)
 		}
 	}
 
-	// use Transform function to remove grafana.app/folder field selector
```

Comment:
Could you please clarify why this isn't needed anymore? Does it have to do with differences between field selectors and label selectors?

Reply:
Yes, instead of using Transform to remove the selectors we use here, we're building newSelector and including only the selectors that we don't use for "special" purposes.  I'll add a comment to the default case above to make that clearer.

Reply:
Ah yes, I see that now. Thanks!